x-pack/plugins/alerts/server/invalidate_pending_api_keys/task.ts

```diff
@@ -0,0 +1,196 @@
+  RunContext,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '../../../task_manager/server';
+import { InvalidateAPIKeyResult } from '../alerts_client';
+import { AlertsConfig } from '../config';
+import { timePeriodBeforeDate } from '../lib/get_cadence';
+import { AlertingPluginsStart } from '../plugin';
+import { InvalidatePendingApiKey } from '../types';
+
+const TASK_TYPE = 'alerts_invalidate_api_keys';
+export const TASK_ID = `Alerts-${TASK_TYPE}`;
+
+const invalidateAPIKey = async (
+  params: InvalidateAPIKeyParams,
+  securityPluginSetup?: SecurityPluginSetup
+): Promise<InvalidateAPIKeyResult> => {
+  if (!securityPluginSetup) {
+    return { apiKeysEnabled: false };
+  }
+  const invalidateAPIKeyResult = await securityPluginSetup.authc.invalidateAPIKeyAsInternalUser(
+    params
+  );
+  // Null when Elasticsearch security is disabled
+  if (!invalidateAPIKeyResult) {
+    return { apiKeysEnabled: false };
+  }
+  return {
+    apiKeysEnabled: true,
+    result: invalidateAPIKeyResult,
+  };
+};
+
+export function initializeApiKeyInvalidator(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>,
+  taskManager: TaskManagerSetupContract,
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  registerApiKeyInvalitorTaskDefinition(
+    logger,
+    coreStartServices,
+    taskManager,
+    config,
+    securityPluginSetup
+  );
+}
+
+export async function scheduleApiKeyInvalidatorTask(
+  logger: Logger,
+  config: Promise<AlertsConfig>,
+  taskManager: TaskManagerStartContract
+) {
+  const interval = (await config).invalidateApiKeysTask.interval;
+  try {
+    await taskManager.ensureScheduled({
+      id: TASK_ID,
+      taskType: TASK_TYPE,
+      schedule: {
+        interval,
+      },
+      state: {},
+      params: {},
+    });
+  } catch (e) {
+    logger.debug(`Error scheduling task, received ${e.message}`);
+  }
+}
+
+function registerApiKeyInvalitorTaskDefinition(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>,
+  taskManager: TaskManagerSetupContract,
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  taskManager.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Invalidate alert API Keys',
+      createTaskRunner: taskRunner(logger, coreStartServices, config, securityPluginSetup),
+    },
+  });
+}
+
+function taskRunner(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>,
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  return ({ taskInstance }: RunContext) => {
+    const { state } = taskInstance;
+    return {
+      async run() {
+        let totalInvalidated = 0;
+        const configResult = await config;
+        try {
+          const [{ savedObjects }] = await coreStartServices;
+          const repository = savedObjects.createInternalRepository([
```

Comment:
We'll need to change this to be an instance of the `SavedObjectsClient` in order to take advantage of encryption